packages/blitz-traits/src/net.rs

```diff
@@ -4,35 +4,44 @@ use std::marker::PhantomData;
 use std::sync::Arc;
 pub use url::Url;
 
-pub type BoxedHandler<D> = Box<dyn RequestHandler<Data = D>>;
+pub type SharedProvider = Arc<dyn NetProvider>;
+pub type BoxedHandler = Box<dyn RequestHandler>;
 pub type SharedCallback<D> = Arc<dyn Callback<Data = D>>;
-pub type SharedProvider<D> = Arc<dyn NetProvider<Data = D>>;
 
 pub trait NetProvider: Send + Sync + 'static {
-    type Data;
-    fn fetch(&self, url: Url, handler: BoxedHandler<Self::Data>);
+    fn fetch(&self, url: Url, handler: BoxedHandler);
 }
 
 pub trait RequestHandler: Send + Sync + 'static {
-    type Data;
-    fn bytes(self: Box<Self>, bytes: Bytes, callback: SharedCallback<Self::Data>);
+    fn bytes(self: Box<Self>, bytes: Bytes);
```

Comment:
Why has this changed (`callback` is no longer passed in by net crate)? I feel like the `Document`/`HtmlDocument`/`HtmlSink` should never need to know about the callbacks.

Reply:
I changed that because it make the API simpler and the callback is only needed when we want to put a resource into the DOM, that does not happen in cases like the `@import` rule.

Reply:
Can you change this back please? It is definitely simpler if the `dom` crate never knows about the callbacks (well, for now, only temporarily in the "resource handlers", but once we make it "url in, bytes out" the `dom` won't need the callbacks at all).